pyfar/plot/line.py

```diff
@@ -104,11 +100,6 @@ def freq(signal, dB=True, log_prefix=20, log_reference=1, xscale='log',
     matplotlib.pyplot.plot() for possible **kwargs.
     """
 
-    if not isinstance(signal, Signal):
-        raise TypeError('Input data has to be of type: Signal.')
-    if xscale not in ['linear', 'log']:
-        raise ValueError(f"xscale must be 'linear' or 'log' but is '{xscale}'")
```

Comment:
Is this check missing in _freq?

Reply:
The checks were inconsistent before. `freq` was the only functions that checked it, also `xscale` is used elsewhere. We also do not check for the `unit` parameter. Should I add those?

Reply:
I think that would be an improvement! Maybe there is an elegant way, the code snippets in `_freq`, `_phase` and `_group_delay` related to `xscale` look pretty similar